### PR TITLE
Add cloudTrail rules [CIS AWS Logging section]

### PR DESCRIPTION
**Pull request by uri-weisman**
Add cloudtrail rules of the logging section.

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

**Review by oren-zohar — DISMISSED** [review]
nice, LGTM

File: bundle/compliance/policy/aws_cloudtrail/verify_s3_object_logging.rego [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	common.array_contains(allowed_types, selector.ReadWriteType)
+
+	dataResource := selector.DataResources[j]
+	dataResource.Type == "AWS::S3::Object"
+
+	partialARN := dataResource.Values[k]
+	partialARN == "arn:aws:s3"
+} else = false {
+	true
+}
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_cloudtrail/verify_s3_object_logging.rego:24]: please run OPA fmt with latest OPA version

File: bundle/compliance/cis_aws/rules/cis_3_11/test.rego [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	eval_fail with input as rule_input(null, true)
+	eval_fail with input as rule_input([], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "WriteOnly"}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "WriteOnly", "DataResources": [{"Type": not_s3_object_type}]}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "ReadOnly", "DataResources": [{"Type": not_s3_object_type}]}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "WriteOnly", "DataResources": [{"Type": s3_object_type, "Values": ["arn:aws:s3"]}]}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "ReadOnly", "DataResources": [{"Type": s3_object_type, "Values": ["arn:aws:s3"]}]}], false)
+}
+
+test_pass {
```
Comment (oren-zohar) [inline bundle/compliance/cis_aws/rules/cis_3_11/test.rego:21]: in the remediation section they mention other fields, are they relevant for the rule evaluation too?
```
aws cloudtrail put-event-selectors --region <region-name> --trail-name <trail-name> --event-selectors '[{ "ReadWriteType": "ReadOnly", "IncludeManagementEvents":true, "DataResources": [{ "Type": "AWS::S3::Object", "Values": ["arn:aws:s3:::<s3-bucket-name>/"] }] }]'
```

File: bundle/compliance/cis_aws/rules/cis_3_10/test.rego [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	eval_fail with input as rule_input(null, true)
+	eval_fail with input as rule_input([], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "ReadOnly"}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "ReadOnly", "DataResources": [{"Type": not_s3_object_type}]}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "WriteOnly", "DataResources": [{"Type": not_s3_object_type}]}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "ReadOnly", "DataResources": [{"Type": s3_object_type, "Values": ["arn:aws:s3"]}]}], true)
+	eval_fail with input as rule_input([{"ReadWriteType": "WriteOnly", "DataResources": [{"Type": s3_object_type, "Values": ["arn:aws:s3"]}]}], false)
+}
+
+test_pass {
```
Comment (oren-zohar) [inline bundle/compliance/cis_aws/rules/cis_3_10/test.rego:21]: in the remediation section they mention other fields, are they relevant for the rule evaluation too?
```
<trail-name> --event-selectors '[{ "ReadWriteType": "WriteOnly", "IncludeManagementEvents":true, "DataResources": [{ "Type": "AWS::S3::Object", "Values": ["arn:aws:s3:::<s3-bucket-name>/"] }] }]'
```

File: bundle/compliance/policy/aws_cloudtrail/esure_no_public_accessibility.rego [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import data.compliance.lib.common
+import data.compliance.policy.aws_cloudtrail.data_adapter
+import future.keywords.every
+
+default no_public_log_access = false
+
+no_public_log_access {
+	grant := data_adapter.trail_bucket_info.acl.Grants[_]
+	not grant.Grantee.URI == "http://acs.amazonaws.com/groups/global/AllUsers"
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_cloudtrail/esure_no_public_accessibility.rego:11]: the benchmark has a different URI
```suggestion
	not grant.Grantee.URI == "https://acs.amazonaws.com/groups/global/AllUsers"
```

File: bundle/compliance/policy/aws_cloudtrail/esure_no_public_accessibility.rego [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+	not grant.Grantee.URI == "http://acs.amazonaws.com/groups/global/AuthenticatedUsers"
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_cloudtrail/esure_no_public_accessibility.rego:12]: same here
```suggestion
	not grant.Grantee.URI == "https://acs.amazonaws.com/groups/global/AuthenticatedUsers"
```

File: bundle/compliance/policy/aws_cloudtrail/ensure_cloudwatch_integration.rego [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package compliance.policy.aws_cloudtrail.ensure_cloudwatch
+
+import data.compliance.lib.common
+import data.compliance.policy.aws_cloudtrail.data_adapter
+
+default ensure_cloudwatch = false
+
+ensure_cloudwatch {
```
Comment (oren-zohar) [inline bundle/compliance/policy/aws_cloudtrail/ensure_cloudwatch_integration.rego:8]: I think the name here should be more specific

File: bundle/compliance/cis_aws/rules/cis_3_6/rule.rego [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+# Ensure S3 bucket access logging is enabled on the CloudTrail S3 bucket
+finding = result {
+	# filter
+	data_adapter.is_single_trail
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(data_adapter.trail_bucket_info.logging.enabled),
+		input.resource,
```
Comment (oren-zohar) [inline bundle/compliance/cis_aws/rules/cis_3_6/rule.rego:14]: can an empty result here be a string? how does it look? what will the evaluation be in that case?

File: bundle/compliance/cis_aws/rules/cis_3_7/rule.rego [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import data.compliance.policy.aws_cloudtrail.data_adapter
+
+# Ensure CloudTrail logs are encrypted at rest using KMS CMKs.
+finding = result {
+	# filter
+	data_adapter.is_single_trail
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(data_adapter.trail.KmsKeyId != null),
```
Comment (oren-zohar) [inline bundle/compliance/cis_aws/rules/cis_3_7/rule.rego:13]: can an empty result here be an empty string?

**Review by oren-zohar — CHANGES_REQUESTED** [review]
Looks good, small comments

File: bundle/compliance/cis_aws/rules/cis_3_11/test.rego (hunk 2 above)
Comment (uri-weisman) [reply bundle/compliance/cis_aws/rules/cis_3_11/test.rego:21]: I don't think we should address `IncludeManagementEvents ` as part of the rule, it's enabled by default ([docs](https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_EventSelector.html))
Furthermore, In rules 3.10 & 3.11 it was not mentioned in the audit section.

File: bundle/compliance/cis_aws/rules/cis_3_10/test.rego (hunk 3 above)
Comment (uri-weisman) [reply bundle/compliance/cis_aws/rules/cis_3_10/test.rego:21]: same as the above comment.

File: bundle/compliance/policy/aws_cloudtrail/esure_no_public_accessibility.rego (hunk 4 above)
Comment (uri-weisman) [reply bundle/compliance/policy/aws_cloudtrail/esure_no_public_accessibility.rego:11]: Good catch 🎖️

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**